src/openlifu/plan/solution.py

```diff
@@ -64,14 +102,221 @@ class Solution:
+
+        Args:
+            transducer: A Transducer item.  #TODO: this should be instanciated at the database level, not here ?
+            options: A struct for solution analysis options.
+
+        Returns: A struct containing the results of the analysis.
+        """
+        solution_analysis = SolutionAnalysis()
+
+        if transducer.id != self.transducer_id:
+            # self.logger.error(f"Provided transducer id {transducer.id} does not match Solution transducer id: {self.transducer_id}")
+            raise ValueError(f"Provided transducer id {transducer.id} does not match Solution transducer id: {self.transducer_id}")
+
+        # dt = 1 / (self.pulse.frequency * 20)
+        # t = self.pulse.calc_time(dt)
+        # input_signal = self.pulse.calc_pulse(t)
+
+        pnp = self.simulation_result.p_min
+        ppp = self.simulation_result.p_max
+        ita = self.simulation_result.ita
+
+        if options.sidelobe_radius is np.nan:
+            options.sidelobe_radius = options.mainlobe_radius
+
+        pnp_MPa = rescale_data_arr(pnp, "MPa")
+
+        # standoff_Z = options.standoff_density * 1500
+        # c_tic = 40e-3  # W cm-1
+        # A_cm = transducer.get_area("cm")
+        # d_eq_cm = np.sqrt(4*A_cm / np.pi)
+        # ele_sizes_cm2 = np.array([elem.get_area("cm") for elem in transducer.elements])
+
+        # xyz = np.stack(np.meshgrid(*coords, indexing="xy"), axis=-1)  #TODO: if fus.Axis is defined, coords.ndgrid(dim="ax")
+        # z_mask = xyz[..., -1] >= options.sidelobe_zmin  #TODO: probably wrong here, should be z{1}>=options.sidelobe_zmin;
+
+        # intensity_Wcm2 = output.intensity.rescale_data("W/cm^2")
+        # pulsetrain_dutycycle = self.get_pulsetrain_dutycycle()
+        # treatment_dutycycle = self.get_treatment_dutycycle()
+        # ita_mWcm2 = self.get_ita(output, units="W/cm^2")
+
+        # power_W = np.zeros(self.num_foci())
+        # TIC = np.zeros(self.num_foci())
+        for focus_index in range(self.num_foci()):
+            if isinstance(self.focus, list):
+                foc = self.focus[focus_index]
+            elif isinstance(self.focus, Point):
+                foc = self.focus
+            # output_signal = []
+            # output_signal = np.zeros((transducer.numelements(), len(input_signal)))
+            # for i in range(transducer.numelements()):
+            #     apod_signal = input_signal * self.apodizations[focus_index, i]
+            #     output_signal[i] = transducer.elements[i].calc_output(apod_signal, dt)
+
+            # p0_Pa = np.max(output_signal)
+
+            # get focus region masks (for mainlobe, sidelobe and beamwidth)
+            mainlobe_mask = mask_focus(
+                self.simulation_result,  #TODO: Original code uses coords, but too complicated to maniplulate a Coordinates class
+                foc,
+                options.mainlobe_radius,
+                mask_op=MaskOp.LESS_EQUAL,
+                units="m",
```

Comment:
These units "m" are the units of the `options.mainlobe_radius`, right? If so I think that the units really should be specified in that `options` object, or at least there should be a docstring under the `mainlobe_radius` attribute of the SolutionOptions class indicating that it must be specified in meters

Reply:
From my understanding, `aspect_ratio` does not have units, it is just a scalar to scale and take into account voxel anisotropy. the `units` here are used to defined to what units all the spatial dimension should be rescaled to.

@peterhollender you coonfirm ?